pkg/signature/kms/azure/client.go

```diff
@@ -304,21 +304,47 @@ func (a *azureVaultClient) createKey(ctx context.Context) (crypto.PublicKey, err
 	return a.public(ctx)
 }
 
-func getKeyVaultSignatureAlgo(algo crypto.Hash) (keyvault.JSONWebKeySignatureAlgorithm, error) {
-	switch algo {
+func (a *azureVaultClient) getKeyVaultHashFunc() (crypto.Hash, error) {
+	publicKey, err := a.public(context.TODO())
```

Comment:
Should we pass the `context` object passed as an argument to the `sign` method down to this `public` call?